### PR TITLE
fs_mitm: Fix mismatched new[] / delete

### PR DESCRIPTION
**Pull request by leoetlino**
Using delete instead of delete[] on a pointer given by new[] is
undefined behaviour.

### CODE REVIEW

**Comment by SciresM** [comment]
If this is really a problem, then this is a problem in a lot of places. Especially in fs.mitm involving romfs paths, which are allocated as new char[]...

Could you make the PR fix every instance of this?

**Comment by m4xw** [comment]
@SciresM this should be fixed.
Should probably fix them step by step if it's deeper in the codebase.

**Comment by SciresM** [comment]
@m4xw I'd rather just fix it in every place all at once, if it's really a problem. fwiw, it seems fine in actual compiled output (IDA shows that it's doing what I'd expect)...but if it's UB then I don't want to do it incrementally. Excise the whole problem.

**Comment by leoetlino** [comment]
Sure, here you go.

**Comment by SciresM** [comment]
Delete[]-ing on memory_source seems incorrect.

For example, we can see [here](https://github.com/Atmosphere-NX/Atmosphere/blob/master/stratosphere/ams_mitm/source/fs_mitm/fsmitm_romfsbuild.cpp#L249) that the RomfsHeader is allocated with new.

However, we can see [here](https://github.com/Atmosphere-NX/Atmosphere/blob/master/stratosphere/ams_mitm/source/fs_mitm/fsmitm_romfsbuild.cpp#L250) that metadata is allocated with new[].

I think we may actually need to switch to malloc/free instead of new/delete, if this is a problem? Man this is dumb.

**Comment by leoetlino** [comment]
Oh sorry, missed the RomfsHeader case...

Yeah hmm it's kind of annoying that this is technically UB. I'll see if there's a nice way to fix this. If not then I'll change it to use malloc/free.

**Comment by TuxSH** [comment]
In the second example, couldn't std::array be used?

**Comment by leoetlino** [comment]
I've replaced new/new[]/delete with malloc/free for memory sources.

I don't think std::array can be used as the size isn't known at compile time.

File: stratosphere/ams_mitm/source/fs_mitm/fsmitm_romfsbuild.cpp [hunk 1]
```diff
@@ -397,13 +399,11 @@ void RomFSBuildContext::Build(std::vector<RomFSSourceInfo> *out_infos) {
     header->dir_table_ofs = header->dir_hash_table_ofs + header->dir_hash_table_size;
     header->file_hash_table_ofs = header->dir_table_ofs + header->dir_table_size;
     header->file_table_ofs = header->file_hash_table_ofs + header->file_hash_table_size;
-    
-    const size_t metadata_size = this->dir_hash_table_size + this->dir_table_size + this->file_hash_table_size + this->file_table_size;
```
Comment (SciresM) [inline stratosphere/ams_mitm/source/fs_mitm/fsmitm_romfsbuild.cpp]: Why remove this?

Comment (SciresM) [reply stratosphere/ams_mitm/source/fs_mitm/fsmitm_romfsbuild.cpp]: Oh, you moved it up. Never mind.

Comment (leoetlino) [reply stratosphere/ams_mitm/source/fs_mitm/fsmitm_romfsbuild.cpp]: I didn't, just moved it up there since the size is needed for malloc anyway

**Pull request merged**